azkaban-common/src/test/java/azkaban/utils/FileIOUtilsTest.java

```diff
@@ -38,16 +39,47 @@
   public TemporaryFolder temp = new TemporaryFolder();
   private File sourceDir, destDir, baseDir;
 
+  /**
+   * Create a very big dir which would cause linux shell command to hard link the dir to
+   * exceed the allowed limit.
+   */
+  private void createBigDir(final String path) throws IOException {
+    final String verylongprefix =
+        "123123123123123123123123123123123113123111111111111111111111111111111111111111111111111111"
+            + "111111111111111111111111111111111111111111111111111111111111111111111111111111111111"
+            + "11231312312312312312313121111111111111111111111111111111111111111111111111";
+
+    for (int i = 1; i <= 400; i++) {
+      final File tmpDir = new File(path + "/" + verylongprefix + "dir" + i);
```

Comment:
Would the new Path API be easier to use than the old File API?

Reply:
my personal feeling is it's slightly more convenient since no string concatenation needs to be done. I replaced a few of them to new API to get the taste of it but not all since there are too much existing usage of old API..